From: Jan Engelhardt <jengelh@inai.de
Date: 2016-03-31 14:13:28.567089312 +0200
References: https://github.com/bji/libs3/pull/38

The upstream Makefile suffers from the typical problems:
* pointless dep stage implemented (though unused)
* impossible to override CFLAGS without breaking the build
* wrong order of LIBS and LDFLAGS
* hardcoded unoverridable directories like "bin"
* lack of using pkg-config
In addition:
* implement export symbols
---
 Makefile.am  |   14 ++++++++++++++
 configure.ac |   13 +++++++++++++
 2 files changed, 27 insertions(+)

Index: libs3-4.0~git257/Makefile.am
===================================================================
--- /dev/null
+++ libs3-4.0~git257/Makefile.am
@@ -0,0 +1,14 @@
+ACLOCAL_AMFLAGS = -I m4
+AM_CPPFLAGS = ${curl_CFLAGS} ${xml_CFLAGS} -Iinc -DLIBS3_VER_MAJOR=\"4\" -DLIBS3_VER_MINOR=\"1\" -DLIBS3_VER=\"4.1\" -D_POSIX_C_SOURCE=200112L
+AM_CFLAGS = -Wall -Wshadow -Wextra
+bin_PROGRAMS = s3
+noinst_PROGRAMS = testsimplexml
+lib_LTLIBRARIES = libs3.la
+s3_SOURCES = src/s3.c
+s3_LDADD = libs3.la
+libs3_la_SOURCES = src/bucket.c src/bucket_metadata.c src/error_parser.c src/general.c src/object.c src/request.c src/request_context.c src/response_headers_handler.c src/service_access_logging.c src/service.c src/simplexml.c src/util.c src/multipart.c
+libs3_la_LDFLAGS = -version-number 4:0:0 -export-symbols-regex '^(S3|simplexml)_'
+libs3_la_LIBADD = -lpthread ${curl_LIBS} ${crypto_LIBS} ${xml_LIBS}
+include_HEADERS = inc/libs3.h
+testsimplexml_SOURCES = src/testsimplexml.c
+testsimplexml_LDADD = libs3.la
Index: libs3-4.0~git257/configure.ac
===================================================================
--- /dev/null
+++ libs3-4.0~git257/configure.ac
@@ -0,0 +1,13 @@
+AC_INIT([libs3], [3.0])
+AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_MACRO_DIR([m4])
+AM_INIT_AUTOMAKE([foreign subdir-objects tar-pax])
+AC_PROG_CC
+AC_DISABLE_STATIC
+LT_INIT
+PKG_CHECK_MODULES([crypto], [libcrypto])
+PKG_CHECK_MODULES([curl], [libcurl])
+PKG_CHECK_MODULES([xml], [libxml-2.0])
+rm -f GNUmakefile
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
